packages/@uppy/companion/src/server/provider/google/drive/index.js

```diff
@@ -192,36 +185,14 @@ class Drive extends Provider {
     })
   }
 
-  logout ({ token }) {
-    return this.#withErrorHandling('provider.drive.logout.error', async () => {
-      await got.post('https://accounts.google.com/o/oauth2/revoke', {
-        searchParams: { token },
-        responseType: 'json',
-      })
-
-      return { revoked: true }
-    })
-  }
-
-  async refreshToken ({ clientId, clientSecret, refreshToken }) {
-    return this.#withErrorHandling('provider.drive.token.refresh.error', async () => {
-      const { access_token: accessToken } = await getOauthClient().post('token', { responseType: 'json', form: { refresh_token: refreshToken, grant_type: 'refresh_token', client_id: clientId, client_secret: clientSecret } }).json()
-      return { accessToken }
-    })
+  // eslint-disable-next-line class-methods-use-this
+  async logout(...args) {
+    return logout(...args)
   }
 
   // eslint-disable-next-line class-methods-use-this
-  async #withErrorHandling (tag, fn) {
-    return withProviderErrorHandling({
-      fn,
-      tag,
-      providerName: Drive.authProvider,
-      isAuthError: (response) => (
-        response.statusCode === 401
-        || (response.statusCode === 400 && response.body?.error === 'invalid_grant') // Refresh token has expired or been revoked
-      ),
-      getJsonErrorMessage: (body) => body?.error?.message,
-    })
+  async refreshToken(...args) {
+    return refreshToken(...args)
   }
 }
```

Comment:
We don't need to disable the linter if we simply define the function on the prototype.
```suggestion
}

Drive.prototype.refreshToken = refreshToken
```

Reply:
I think this is gonna cause problems once we port this to TS

Reply:
I suggest we either don't care about it until/if this happens, and/or we use an intermediary parent class as suggested.

Reply:
i used a parent class initially but @Murderlon didn't like it https://github.com/transloadit/uppy/pull/5061/commits/08beacc4be88762d257f0d6f73bad3bb81bb2a51 https://github.com/transloadit/uppy/pull/5061#discussion_r1598040585

Reply:
So let's add the methods to the prototype, no?